File: easy/1848-minimum-distance-to-the-target-element/solution.py
from typing import List


class Solution:
    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        """
        1848. Minimum Distance to the Target Element
        Time: O(n)
        Space: O(1)
        """
        min_dist = float('inf')

        for i, num in enumerate(nums):
            if num == target:
                min_dist = min(min_dist, abs(i - start))

        return min_dist

File: easy/1848-minimum-distance-to-the-target-element/solution.java
/**
 * 1848. Minimum Distance to the Target Element
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int getMinDistance(int[] nums, int target, int start) {
        int minDist = Integer.MAX_VALUE;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == target) {
                minDist = Math.min(minDist, Math.abs(i - start));
            }
        }

        return minDist;
    }
}
